### PR TITLE
Refactored Room into states. Added black background to HUD to draw overtop of Room.

### PR DESCRIPTION
**Pull request by NickJohn547745**

### CODE REVIEW

**Review by eskripac — APPROVED** [review]
Looks good, good stuff

File: Sprint0/RoomStateClasses/ARoomTransitionState.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    public override void Reset()
+    {
+        ReturnToIdle();
+    }
+
+    public void ReturnToIdle()
+    {
+        nextRoomState.roomOffset = new Point();
```
Comment (Will-Blanton) [inline Sprint0/RoomStateClasses/ARoomTransitionState.cs:62]: The abstract class was a good choice for reducing duplicate code and for following our standard practices.

File: Sprint0/RoomStateClasses/RoomIdleState.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+namespace sprint0.RoomStateClasses;
+
+public class RoomIdleState : ARoomState
+{
+
+    public RoomIdleState(RoomStateManager state)
+    {
+        roomState = state;
+
+        roomState.game.CollisionManager.Reset();
```
Comment (Will-Blanton) [inline Sprint0/RoomStateClasses/RoomIdleState.cs:24]: RoomIdleState is a big improvement to the previous room classes. It is much more readable due to its concise methods

File: Sprint0/RoomStateClasses/RoomStateManager.cs [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        }
+    }
+
+    private void InitializeAdjacentLevels()
+    {
+        for (int i = 0; i < 4; i++)
+        {
+            int currentDestination = levelConfig.Destinations[i];
+
+            if (currentDestination >= 0)
```
Comment (Will-Blanton) [inline Sprint0/RoomStateClasses/RoomStateManager.cs:87]: RoomState manager was a good choice for handling all of the level config initialization logic. The only change here would be to modify it to include the procedural rooms here, which we have already talked about.

File: Sprint0/RoomStateClasses/RoomTransitionRightState.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            }
+        } else
+        {
+            if (!isIncoming)
+            {
+                roomState = nextRoomStateManager;
+                roomState.currentState = new RoomIdleState(nextRoomStateManager);
+                roomState.game.gameState.RoomState = roomState;
+
+                DungeonMap.Instance.AddRoomToMap(Direction.Right);
```
Comment (Will-Blanton) [inline Sprint0/RoomStateClasses/RoomTransitionRightState.cs:53]: There may be a way to only have one direction transition class by having a set of fields that you can initialize differently depending on the direction, but this is still an improvement to the previous logic and has fixed a significant amount of errors with room transitions

**Review by Will-Blanton — COMMENTED** [review]
Looks really good to me. Refactoring this has made our code much cleaner. I left other more specific comments for certain files with a couple of possible changes.

File: Sprint0/DoorClasses/Door.cs [hunk 5]
```diff
@@ -20,7 +21,10 @@ public abstract class Door : ICollidable
         public void Collide(ICollidable obj, ICollidable.Edge edge)
         {
             if (obj.Type == ICollidable.ObjectType.Player && (Id == 1 || Id == 4))
+            { 
                 HasCollided = true;
+                Debug.WriteLine("PreCOLLLLL");
```
Comment (natdrog) [inline Sprint0/DoorClasses/Door.cs:26]: If this line is for debugging, you should probably remove it before merging the pull request

**Review by natdrog — APPROVED** [review]
I like the new transition classes and managers. Helps reduce some complexity from just having one large class

**Review by Awilkes21 — APPROVED** [review]

**Pull request closed without merging**